Source/JavaScriptCore/assembler/MacroAssemblerARM64.h

```diff
@@ -913,6 +913,12 @@ class MacroAssemblerARM64 : public AbstractMacroAssembler<Assembler> {
         m_assembler.lsl<32>(dest, src, imm.m_value & 0x1f);
     }
 
+    void lshift32(TrustedImm32 imm, RegisterID shiftAmount, RegisterID dest)
+    {
+        move(imm, getCachedDataTempRegisterIDAndInvalidate());
```

Comment:
Can we avoid materializing the immediate when the immediate is small enough to use the immediate form of LSL?